datatable/java/datatable/src/main/java/io/cucumber/datatable/DataTableType.java

```diff
@@ -78,6 +81,51 @@ private DataTableType(Class<?> type, Type target, RawTableTransformer<?> transfo
         this(type, aListOf(aListOf(type)), new TableCellTransformerAdaptor<>(transformer));
     }
 
+    /**
+     * Creates a data table type that transforms the cells of the table into a list of objects,
+     * using table headers to set field/property names on the objects.
+     * <p>
+     * The class needs an empty constructor, and fields must have public setters (or the fields themselves must be public).
+     *
+     * @param type the type of the entry
+     * @param <T>  see <code>type</code>
+     */
+    public static <T> DataTableType entry(final Class<T> type) {
+        return new DataTableType(type, new TableEntryTransformer<T>() {
+            @Override
+            public T transform(Map<String, String> entry) {
+                ObjectMapper objectMapper = new ObjectMapper();
+                return objectMapper.convertValue(entry, type);
+            }
+        });
+    }
+
+    /**
+     * Creates a data table type that transforms individual cells to another type, calling the type's
+     * String constructor.
+     *
+     * @param type the type of the cell
+     * @param <T> see <code>type</code>
+     * @throws CucumberDataTableException if <code>type</code> does not has a public String constructor.
+     */
+    public static <T> DataTableType cell(final Class<T> type) {
+        try {
+            final Constructor<T> constructor = type.getConstructor(String.class);
+            return new DataTableType(type, new TableCellTransformer<T>() {
+                @Override
+                public T transform(String cell) {
+                    try {
+                        return constructor.newInstance(cell);
```

Comment:
To the best of my knowledge `objectMapper.convertValue("someThing", SingleArgConstructor.class);` works. So you use Jackson here as well. I believe it is also a bit more flexible as it will check for methods name "fromString" as well.


 It simplifies the documentation (refer to Jackons)